### PR TITLE
WIP: Dharma Loans for loan requests

### PR DESCRIPTION
**Pull request by onggunhao**
Opening a WIP PR to track changes and discussions

### CODE REVIEW

File: ethereum/contracts/core/EnableContractRegistry.sol [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    event ContractAddressUpdated(
+        ContractType indexed contractType,
+        address indexed oldAddress,
+        address indexed newAddress
+    );
+
+    enum ContractType {
+        PermissionsLib,
+        StudentLoanCrowdfundFactory
```
Comment (adibas03) [inline ethereum/contracts/core/EnableContractRegistry.sol]: Do we need *StudentLoanCrowdfundFactory* , why not just generic *LoanCrowdfundFactory* ,
and the details of the loan would identify what specific type of loan ?

File: ethereum/contracts/crowdloan/StudentLoanCrowdfund.sol [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    //This works like a mintable token, supply starts at zero. If and when we make it an actual token we can replace this.
+    mapping (address => uint) debtTokens;
+    uint debtTokensSupply;
+
+    enum LoanStatus {
+        NOT_STARTED,
+        PARTIALLY_FUNDED,
+        FULLY_FUNDED,
+        STARTED
```
Comment (adibas03) [inline ethereum/contracts/crowdloan/StudentLoanCrowdfund.sol]: should there be a `FULLY REPAID` as well as `REPAYING`

File: ethereum/contracts/crowdloan/StudentLoanCrowdfund.sol [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    IERC20 debtToken;
+    IERC20 loanToken;
+
+    event AddFunding(address indexed lender, uint indexed amount);
+    event RevokeFunding(address indexed lender, uint indexed amount);
+    event WithdrawPayment(address indexed lender, uint indexed amount);
+
+    modifier onlyDebtHolder() {
+        require(debtToken[msg.sender] > 0, NO_DEBT_TOKENS);
```
Comment (adibas03) [inline ethereum/contracts/crowdloan/StudentLoanCrowdfund.sol]: should be
`require(debtTokens[msg.sender] > 0, NO_DEBT_TOKENS);`
*debtTokens*

File: ethereum/contracts/crowdloan/StudentLoanCrowdfund.sol (hunk 2 above)
Comment (tspoff) [reply ethereum/contracts/crowdloan/StudentLoanCrowdfund.sol]: Good point - I'll add these.

File: ethereum/contracts/LoanRequest.sol [hunk 4]
```diff
@@ -13,7 +13,7 @@ import "./TokenTransfer.sol";
     The fallback function just refunds - we don't accept ether here. (Need to evaluate re-entrancy)
 
     So we may have to override huge parts of it, but we'll see
-*/
+*/oma
```
Comment (adibas03) [inline ethereum/contracts/LoanRequest.sol]: typo?

File: ethereum/contracts/core/EnableContractRegistry.sol (hunk 1 above)
Comment (onggunhao) [reply ethereum/contracts/core/EnableContractRegistry.sol]: Makes sense!

**Pull request merged**